src/main/java/net/smartcosmos/dao/relationships/impl/RelationshipPersistenceService.java

```diff
@@ -32,237 +39,284 @@
+            exceptionLogger("RelationshipPersistenceService:findByTypeForSource", e, tenantUrn, sourceType, sourceUrn, relationshipType);
         }
-
-        return convert(entityList);
+        return responsePage;
     }
 
-    @Deprecated
-    public List<RelationshipResponse> findByType(String tenantUrn, String entityReferenceType, String referenceUrn, String type) {
 
-        UUID accountId = UuidUtil.getUuidFromUrn(tenantUrn);
+    @Override
+    public Page<RelationshipResponse> findByTypeForTarget(
+        String tenantUrn,
+        String targetType,
+        String targetUrn,
+        String relationshipType,
+        Integer page,
+        Integer size,
+        SortOrder sortOrder,
+        String sortBy) {
+
+        Pageable pageable = buildPageable(page, size, sortOrder, sortBy);
+
+        Page<RelationshipResponse> responsePage = RelationshipPersistenceUtil.emptyPage();
 
-        List<RelationshipEntity> entityList = new ArrayList<>();
         try {
-            entityList = relationshipRepository.findByTenantIdAndSourceTypeAndSourceIdAndRelationshipType(
-                accountId,
-                entityReferenceType,
-                UuidUtil.getUuidFromUrn(referenceUrn),
-                type);
+            org.springframework.data.domain.Page<RelationshipEntity> entityPage =
+                relationshipRepository.findByTenantIdAndTargetTypeAndTargetIdAndRelationshipType(
+                    UuidUtil.getUuidFromUrn(tenantUrn),
+                    targetType,
+                    UuidUtil.getUuidFromUrn(targetUrn),
+                    relationshipType,
+                    pageable);
+
+            return conversionService.convert(entityPage, responsePage.getClass());
+
         } catch (IllegalArgumentException e) {
-            // empty list will be returned anyway
-            log.warn("Illegal URN submitted by account %s: reference URN %s", tenantUrn, referenceUrn);
+            exceptionLogger("RelationshipPersistenceService:findByTypeForTarget", e, tenantUrn, targetType, targetUrn, relationshipType);
         }
-
-        return convert(entityList);
+        return responsePage;
     }
+    // endregion
 
-    @Deprecated
-    public List<RelationshipResponse> findByTypeReverse(String tenantUrn, String relatedEntityReferenceType, String relatedReferenceUrn, String type) {
+    // region Find All
+    @Override
+    public Page<RelationshipResponse> findAllForSource(
+        String tenantUrn,
+        String sourceType,
+        String sourceUrn,
+        Integer page,
+        Integer size,
+        SortOrder sortOrder,
+        String sortBy) {
 
-        UUID accountId = UuidUtil.getUuidFromUrn(tenantUrn);
+        Pageable pageable = buildPageable(page, size, sortOrder, sortBy);
+
+        Page<RelationshipResponse> responsePage = RelationshipPersistenceUtil.emptyPage();
```

Comment:
There is no reason to have this returned to a local.  Simply call the method in the repository method below.
```
            org.springframework.data.domain.Page<RelationshipEntity> entityPage =
                relationshipRepository.findByTenantIdAndSourceTypeAndSourceIdAndTargetTypeAndTargetId(
                UuidUtil.getUuidFromUrn(tenantUrn),
                sourceType,
                UuidUtil.getUuidFromUrn(sourceUrn),
                targetType,
                UuidUtil.getUuidFromUrn(targetUrn),
                buildPageable(page, size, sortOrder, sortBy));
```